lib/nerves_hub_link/archive_manager.ex

```diff
@@ -0,0 +1,148 @@
+defmodule NervesHubLink.ArchiveManager do
+  @moduledoc """
+  GenServer responsible for brokering messages between:
+    * an external controlling process
+    * HTTP
+
+  Should be started in a supervision tree
+  """
+  use GenServer
+
+  alias NervesHubLink.Client
+  alias NervesHubLink.Downloader
+  alias NervesHubLink.Message.ArchiveInfo
+
+  require Logger
+
+  defmodule State do
```

Comment:
We've switched to not use `State` modules any more for GenServer state. It's mostly stylistic, but there is a cost on the slower devices to every `.beam` file that gets loaded.

Reply:
Huh, I copied from the update manager so I went with what that had. Should I remove the state module from the other manager as a separate PR then?